web/src/lightbox.js

```diff
@@ -327,7 +327,10 @@ export function build_open_media_function(on_close) {
         // if the asset_map already contains the metadata required to display the
         // asset, just recall that metadata.
         let $preview_src = $media.attr("src");
-        let payload = asset_map.get($preview_src);
+        let payload = asset_map.get([
+            $preview_src,
+            $media.parent().attr("aria-label") || $media.parent().attr("href"),
+        ]);
```

Comment:
The logic seems potentially correct, but I think we should make a function `get_asset_map_key_for_media($preview_src, $media)` that does this calculation, rather than copy/pasting it in 3 places.

Reply:
@timabbott I have made the function and modified the code.